api/src/test/java/com/epam/pipeline/manager/docker/DockerRegistryNotificationTest.java

```diff
@@ -16,326 +16,398 @@
-import static org.mockito.Matchers.anyString;
-import static org.mockito.Mockito.doNothing;
-import static org.mockito.Mockito.when;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
 
 
-@DirtiesContext
-@ContextConfiguration(classes = TestApplicationWithAclSecurity.class)
-public class DockerRegistryNotificationTest extends AbstractManagerTest {
+public class DockerRegistryNotificationTest extends AbstractAclTest {
 
     private static final String ADMIN = "ADMIN";
-    private static final String TEST_USER = "USER";
-    private static final String TEST_REPO = "repository";
-    private static final String TEST_REPO_WITH_EXTERNAL_PATH = "repository3";
-    private static final String EXTERNAL_REPO_PATH = "external_repository";
-    private static final String TEST_REPO_WITHOUT_WRITE_ACCESS = "repository2";
-    private static final String TEST_IMAGE = "library/image";
-    private static final String TEST_IMAGE_NEW_GROUP = "library2/image";
+    private static final String USER = "USER";
+    private static final String REPO = "repository";
+    private static final String REPO_EXTERNAL_PATH = "repository2";
+    private static final String EXTERNAL_PATH = "external_repository";
+    private static final String REPO_NO_WRITE_ACCESS = "repository3";
+    private static final String IMAGE_GROUP_1 = "library";
+    private static final String IMAGE_GROUP_2 = "library2";
+    private static final String IMAGE = IMAGE_GROUP_1 + "/image";
+    private static final String IMAGE_NEW_GROUP = IMAGE_GROUP_2 + "/image";
     private static final String PUSH_ACTION = "push";
     private static final String PULL_ACTION = "pull";
-    private static final String ADMIN_ROLE = "ADMIN";
-    public static final String LATEST = "latest";
-    public static final long DOCKER_SIZE = 123456L;
+    private static final String LATEST = "latest";
+    private static final String TEST_IDENTIFIER = REPO + "/" + IMAGE_GROUP_1;
+    private static final ImmutablePair<String, String> IMMUTABLE_PAIR_1 = new ImmutablePair(IMAGE_GROUP_1, "image");
+    private static final ImmutablePair<String, String> IMMUTABLE_PAIR_2 = new ImmutablePair(IMAGE_GROUP_2, "image");
+    private static final Long REG_ID_1 = 1L;
+    private static final Long REG_ID_2 = 2L;
+    private static final Long REG_ID_3 = 3L;
+    private static final Long TOOLGROUP_ID_1 = 4L;
+    private static final Long TOOL_ID_1 = 5L;
+    private static final Long TOOL_ID_2 = 6L;
+    private static final Long TOOL_ID_3 = 7L;
+    private static final Long CONTEXT_ID = 8L;
+    private static final Long ROLE_ID = 9L;
+    private static final CharSequence NOT_ALLOWED_MESSAGE = String.format("Permission is not granted for '%s' on '%S'",
+            REPO_NO_WRITE_ACCESS, AclPermission.WRITE_NAME);
+    private static final String ROLE_USER = "ROLE_USER";
```

Comment:
Could we reuse already existing constants from `AbstractAclTest`?

Reply:
Done!